sklearn/utils/fixes.py

```diff
@@ -107,22 +108,25 @@ def _eigh(*args, **kwargs):
 
 
 # remove when https://github.com/joblib/joblib/issues/1071 is fixed
-def delayed(function):
+def delayed(func, thread=threading.current_thread()):
```

Comment:
Is the `thread` kwarg here only used for testing? I'm concerned about how `delayed` is in `utils.fixes`, which is "kind of public".

If we want to be careful and not introduce more API:

```python
def delayed(func):
    return _delayed(func)

def _thread(func, thread=threading.current_thread()):
    # same as this PR, but use `thread` for testing.
```

Reply:
I guess you meant:

```python
def delayed(func):
    return _delayed(func)

def _delayed(func, thread=threading.current_thread()):
    # same as this PR, but use `thread` for testing.
```

Reply:
Indeed this is something we did not like when investigating with @fcharras

Reply:
Yea, I meant `_delayed` and using that function for testing.